dali/dfuplus/dfuplus.cpp

```diff
@@ -1518,16 +1518,16 @@ int CDfuPlusHelper::listhistory()
             out.append("<History>").newline();
             for(unsigned i = 0; i < arrHistory.length(); i++)
             {
-                IConstFileHistory* historyRecord = &arrHistory.item(i);
+                IConstHistory* historyRecord = &arrHistory.item(i);
                 if(historyRecord != NULL)
                 {
-                    out.appendf("\t<Origin ip=\"%s\"", historyRecord->getOriginIP()).newline();
-                    out.appendf("\t\tname=\"%s\"", historyRecord->getOriginName()).newline();
-                    out.appendf("\t\toperation=\"%s\"", historyRecord->getOriginOperation()).newline();
-                    out.appendf("\t\towner=\"%s\"", historyRecord->getOriginOwner()).newline();
-                    out.appendf("\t\tpath=\"%s\"", historyRecord->getOriginPath()).newline();
-                    out.appendf("\t\ttimestamp=\"%s\"/>", historyRecord->getOriginTimestamp()).newline();
-                    out.appendf("\t\tworkunit=\"%s\"/>", historyRecord->getOriginWorkunit()).newline();
+                    out.appendf("\t<Origin ip=\"%s\"", historyRecord->getIP()).newline();
+                    out.appendf("\t\tname=\"%s\"", historyRecord->getName()).newline();
```

Comment:
since you're already doing an appendf, it might be more efficient to add the new line in the formatted string as opposed to calling the newline method.
out.appendf("\t\tname=\"%s\"\n", historyRecord->getName());